includes/class-wc-gateway-dintero-hp.php

```diff
@@ -1361,11 +1358,16 @@ public function callback( $return_page = false ) {
 			if ( ! empty( $order ) && $order instanceof WC_Order ) {
 				$amount = absint( strval( floatval( $order->get_total() ) * 100 ) );
 				if ( array_key_exists( 'status', $transaction ) &&
-					 array_key_exists( 'amount', $transaction ) &&
-					 $transaction['amount'] === $amount ) {
-
-					if ( 'AUTHORIZED' === $transaction['status'] ) {
-
+					 array_key_exists( 'amount', $transaction )) {
+
+					if ($transaction['amount'] < $amount - 1 ) {
```

Comment:
Should I also check put on-hold if `transaction['amount']` is greater than `amount`?